controllers/workspace/metrics/metrics.go

```diff
@@ -21,14 +21,26 @@ const (
 	workspaceSourceLabel     = "controller.devfile.io/devworkspace-source"
 	metricSourceLabel        = "source"
 	metricsRoutingClassLabel = "routingclass"
+	metricsReasonLabel       = "reason"
+
+	reasonInfrastructureFailure = "infrastructure_failure"
+	reasonUnknown               = "unknown"
 )
 
+var infrastructureFailures = []string{
+	"CrashLoopBackOff",
```

Comment:
Well, it's not a necessary infrastructure error. I would say in most cases it's caused by Che bug or a user's mistake.